<?xml version="1.0"?>
<launch>
  <!-- GDB functionality -->
  <arg name="debug" default="false" />
  <arg unless="$(arg debug)" name="launch_prefix" value="" />
  <arg     if="$(arg debug)" name="launch_prefix" value="gdb --ex run --args" />

  <!-- Load example URDF -->
  <param name="robot_description" command="$(find xacro)/xacro '$(find turtlebot3_description)/urdf/turtlebot3_burger.urdf.xacro'" />

    <include file="$(find rplidar_ros)/launch/rplidar.launch"/>
    <node pkg="usb_cam" type="usb_cam_node" name="usb_cam" output="log"/>

    <!-- Load controller settings -->
    <rosparam file="$(find medicbot_control)/config/medicbot_diff_drive_controller.yaml" command="load"/>

    <remap from="base_controller/odom" to="odom"/>	
    <remap from="base_controller/cmd_vel" to="cmd_vel"/>	
    <!-- Load hardware interface -->
    <node name="medicbot_hw_interface_node" pkg="medicbot_control" type="medicbot_hw_interface_node"
          output="screen"  launch-prefix="$(arg launch_prefix)"/>

    <!-- Load controller manager -->
    <node name="ros_control_controller_manager" pkg="controller_manager" type="controller_manager" respawn="false" output="log" args="spawn joint_state_controller base_controller" />

    <!-- Convert joint states to /tf tranforms -->
    <!--node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher">
	    <param name="publish_frequency" type="double" value="50.0" />
    </node-->
</launch>
